### DISCUSSION TITLE
Confusion regarding Look At Modes with Third Person Follow

### DISCUSSION THREAD

**Discussion by AcevedoC17** [ramokz/phantom-camera#148] [Question / Help Wanted] [answered]
I wish to make the camera orbit around the selector, but I seem to be misunderstanding the look at modes. The camera refuses to orbit around the selector node properly, it seems to ignore the fact that it should be looking at it at all times. I am certain I am doing something wrong here, but I cannot figure out the issue. 

For the camera movement, I followed the example presented in the docs under Example Setup under Third Person Follow

https://github.com/ramokz/phantom-camera/assets/77506847/a83b0be6-5e8f-436d-b152-1425c8255e23

**Comment by ramokz** [comment]
The combination of `Third Person Follow` and `Look At Modes` isn't working nicely together at the minute. Mainly due to they're partially mutually exclusive in how they work, which hasn't been properly addressed yet.
But that _should_ be irrelevant in your case, as `Third Person Follow` will make the `PCam` always point towards its given target, which I can understand is what you're looking to do? So it might be worth not setting a `Look At Mode` and just solely rely on `Third Person Follow`.

Looking at your video example, it seems like the position of the follow target, which I'm guessing is an invisible `Node3D`(?), doesn't match the selected/highlighted node's position in the scene. Again, that might be related to `Look At Mode` + `Third Person Follow`.

**Reply by AcevedoC17** [reply]
I went ahead and tested Third Person Follow without any look at mode and the behavior is the exact same. You're correct, the follow target is an invisible Node3D which essentially always has the same position as the visible selector. Moving the Node3D around in the 3d editor makes the camera follow it as expected, but whenever any rotation is involved, the camera ignores keeping the Node3D focused. I can't seem to replicate the behavior observed in the examples this way. Basically, the follow part works fine, but the "orbit" functionality is giving me problems.

**Reply by ramokz** [reply]
That sounds very odd.

If you add a visible mesh as a child of the invisible `Node3D` target, with a relative position of `0,0,0`, does the centre of the orbital rotation look correct based on that position, or is it still entirely off?

**Reply by AcevedoC17** [reply, marked as answer]
I've actually figured out the issue. It was an incorrect value in the Target Offset of the PhantomCamera3D. This fixes the issue, but now I'm presented with another problem which I'm not entirely sure is fixable. The best camera mode for my situation is Orthogonal, but using this camera mode with Third Person Follow causes a lot of clipping even when making the spring length very large. Was the third person follow mode made with Perspective Camera in mind or should it be fine with other modes as well?

**Reply by AcevedoC17** [reply]
For anyone wondering, Third Person Follow works well with Orthogonal Camera, the issue with clipping can be solved by using 

set_orthogonal(size, z_near, z_far) z_near to a large negative value.

**Reply by ramokz** [reply]
I remember doing a few follow mode tests in an `Orthogonal` view, but `Third Person Follow` was made with a `Perspective` view in mind, yes.

But your solve makes sense, as it sounds like it just needs a much larger distance and camera culling value to avoid the clipping problem.

Thanks for sharing your findings!
